### PR TITLE
fixes to edit user privileges

### PR DESCRIPTION
**Pull request by briri**
#794 
- Org admin users no longer see the 'Edit' link for Super Admins
- Org admins can now only grant/revoke privileges that they themselves have access to

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -11,9 +11,9 @@ class UsersController < ApplicationController
   def admin_index
     authorize User
     if current_user.can_super_admin?
-      @users = User.includes(:roles).page(1)
+      @users = User.includes(:perms).page(1)
```
Comment (jollopre) [inline app/controllers/users_controller.rb]: the User model has default_scope including :org and :perms already so is not really needed to include :perms there

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -22,17 +22,28 @@ def admin_index
...
+    if current_user.can_super_admin?
+      perms = Perm.all
+    else
+      perms = current_user.perms
+    end
+    #perms = user_perms & [Perm.grant_permissions, Perm.modify_templates, Perm.modify_guidance, 
+    #                       Perm.use_api, Perm.change_org_details, Perm.add_orgs, 
+    #                       Perm.change_affiliation, Perm.grant_api]
+    
+puts perms.collect(&:name)
```
Comment (jollopre) [inline app/controllers/users_controller.rb]: please remove this puts

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -22,17 +22,28 @@ def admin_index
...
-                           Perm.change_affiliation, Perm.grant_api]
+    
+    # Super admin can grant any Perm, org admins can only grant Perms they
+    # themselves have access to
+    if current_user.can_super_admin?
+      perms = Perm.all
+    else
+      perms = current_user.perms
+    end
+    #perms = user_perms & [Perm.grant_permissions, Perm.modify_templates, Perm.modify_guidance,
```
Comment (jollopre) [inline app/controllers/users_controller.rb]: remove the code please, we have git :)

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -22,17 +22,28 @@ def admin_index
...
-    @perms = user_perms & [Perm.grant_permissions, Perm.modify_templates, Perm.modify_guidance, 
-                           Perm.use_api, Perm.change_org_details, Perm.add_orgs, 
-                           Perm.change_affiliation, Perm.grant_api]
+    
+    # Super admin can grant any Perm, org admins can only grant Perms they
+    # themselves have access to
+    if current_user.can_super_admin?
+      perms = Perm.all
+    else
+      perms = current_user.perms
```
Comment (jollopre) [inline app/controllers/users_controller.rb]: you have user_perms above referencing the same thing

File: app/views/users/_admin_grant_permissions.html.erb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<% namesHash = name_and_text %>
+<%# namesHash = name_and_text %>
```
Comment (jollopre) [inline app/views/users/_admin_grant_permissions.html.erb]: please remove it if no longer applicable

File: app/views/users/_admin_grant_permissions.html.erb [hunk 6]
```diff
@@ -10,32 +10,32 @@
         <h1 class="modal-title"><%= _('Edit user privileges') %></h1>
       </div>
       
-      <%= form_tag( admin_update_permissions_user_path(@user), method: :put, remote: true, class: 'admin_update_permissions' ) do %>
+      <%= form_tag( admin_update_permissions_user_path(user), method: :put, remote: true, class: 'admin_update_permissions' ) do %>
```
Comment (jollopre) [inline app/views/users/_admin_grant_permissions.html.erb]: nice to see plain variables instead of instance ones

**Comment by briri** [comment]
Made suggested changes. Also rebased to pull in the latest from development

**Pull request merged**